### DISCUSSION TITLE
TypeError: SOLOv2: __init__() got an unexpected keyword argument 'roi_head'

### DISCUSSION THREAD

**Discussion by yasarniyazoglu** [open-mmlab/mmdetection#9720] [Q&A] [answered]
from mmdet.apis import set_random_seed
from mmcv import Config
cfg = Config.fromfile('/content/mmdetection/config.py')
cfg.model.roi_head.bbox_head.num_classes = 3
cfg.model.roi_head.mask_head.num_classes = 3
cfg.data.samples_per_gpu = 2
cfg.work_dir = './sonuc_exps'
cfg.optimizer.lr = 0.02 / 8
cfg.lr_config.warmup = None
cfg.log_config.interval = 10
cfg.runner.max_epochs = 10
cfg.evaluation.metric = 'mAP'
cfg.evaluation.metric = 'bbox'
cfg.lr_config.policy = 'step'
cfg.lr_config.step = 2
cfg.evaluation.interval = 20
cfg.checkpoint_config.interval = 20
cfg.load_from = '/content/mmdetection/checkpoints/solov2_x101_dcn_fpn_3x_coco_20220513_214337-aef41095.pth'
cfg.seed = 0
set_random_seed(0, deterministic=False)
cfg.device = 'cuda'
cfg.gpu_ids = range(1)
cfg.log_config.hooks = [
    dict(type='TextLoggerHook'),
    dict(type='TensorboardLoggerHook')]

print(f'Config:\n{cfg.pretty_text}')

from mmdet.datasets import build_dataset
from mmdet.models import build_detector
from mmdet.apis import train_detector

datasets = [build_dataset(cfg.data.train)]
model = build_detector(cfg.model)
model.CLASSES = datasets[0].CLASSES
mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))
train_detector(model, datasets, cfg, distributed=False, validate=True)
# Error
loading annotations into memory...
Done (t=0.01s)
creating index...
index created!
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[/usr/local/lib/python3.8/dist-packages/mmcv/utils/registry.py](https://localhost:8080/#) in build_from_cfg(cfg, registry, default_args)
     68     try:
---> 69         return obj_cls(**args)
     70     except Exception as e:

TypeError: __init__() got an unexpected keyword argument 'roi_head'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
4 frames
[/usr/local/lib/python3.8/dist-packages/mmcv/utils/registry.py](https://localhost:8080/#) in build_from_cfg(cfg, registry, default_args)
     70     except Exception as e:
     71         # Normal TypeError does not print class name.
---> 72         raise type(e)(f'{obj_cls.__name__}: {e}')
     73 
     74 

TypeError: SOLOv2: __init__() got an unexpected keyword argument 'roi_head'

**Comment by ZwwWayne** [comment, marked as answer]
SOLOv2 does not have RoIHead, therefore, you do not need to set
```
cfg.model.roi_head.bbox_head.num_classes = 3
cfg.model.roi_head.mask_head.num_classes = 3
```
Instead, you should set
```
cfg.model.mask_head.num_classes = 3
```
You should modify the keys existing in the config of solov2
